test/export/note_sorter_spec.py

```diff
@@ -0,0 +1,119 @@
+from mamba import describe, it, context, before
+from unittest.mock import MagicMock
+from random import shuffle
+
+from test_utils.anki import mock_anki_modules
+
+mock_anki_modules()
+
+from crowd_anki.export.note_sorter import NoteSorter
+from crowd_anki.config.config_settings import ConfigSettings
+
+
+test_guids = ["abc", "bcd", "cde", "def", "efg", "fgh"]
+test_flags = [0, 1, 2, 3, 4, 5]
+test_tags = ["adjectives", "directions", "interesting", "nouns", "verbs", "zzzzFinal"]
+test_notemodels = ["Default", "LL Noun", "LL Sentence", "LL Verb", "LL Word", "Zulu"]
+test_notemodelids = test_guids
+test_fields = test_tags
+
+
+class NoteSorterTester():
+    def __init__(self):
+        self.notes = self.setup_notes()
+
+        self.config = ConfigSettings(False)
+        self.config.export_deck_sort_reversed = False
+
+    def get_single_note_mock(self, i):
+        note = MagicMock()
+
+        note.anki_object.guid = test_guids[i]
+        note.anki_object.flags = test_flags[i]
+        note.anki_object.tags = test_tags[i]
+
+        note.anki_object._model = {
+            "name": test_notemodels[i],
+            "crowdanki_uuid": test_notemodelids[i]
+        }
+
+        note.anki_object.fields = [test_fields[i], test_fields[i]]
+
+        return note
+
+    def setup_notes(self):
+        random_range = list(range(0, len(test_guids)))
+        shuffle(random_range)
+        notes_list = [self.get_single_note_mock(i) for i in random_range]
+
+        return notes_list
+    
+    def set_sort_method(self, set_to):
+        self.config.export_deck_sort_methods = set_to
+    
+    def setup(self):
```

Comment:
ah, this actually does sorting. How about another new name 😛  - `sort_with([sort_methods])` to make it clear that we actually run the sort here with the given sort method